### PR TITLE
Feature/bump clion

### PR DESCRIPTION
**Pull request by strahlc**
dev-util/clion: version bump to 2018.3
dev-util/clion: remove old version

### CODE REVIEW

File: dev-util/clion/clion-2018.2.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
```
Comment (juippis) [inline dev-util/clion/clion-2018.2.6.ebuild]: Looks like 7 can be used.

File: dev-util/clion/clion-2018.2.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+DESCRIPTION="A complete toolset for C and C++ development"
+HOMEPAGE="http://www.jetbrains.com/clion"
+SRC_URI="http://download.jetbrains.com/cpp/CLion-${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="IDEA
+	|| ( IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal )"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+RESTRICT="splitdebug"
+IUSE="-custom-jdk"
```
Comment (juippis) [inline dev-util/clion/clion-2018.2.6.ebuild]: How is `-custom-jdk` different from `custom-jdk`? Legit question, not complaint.

File: dev-util/clion/clion-2018.2.6.ebuild [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+src_install() {
+	local dir="/opt/${P}"
+
+	insinto "${dir}"
+	doins -r *
+	fperms 755 "${dir}"/bin/{clion.sh,fsnotifier{,64},clang/linux/clang{d,-tidy}}
+
+	if use custom-jdk; then
+		if [[ -d jre64 ]]; then
+		fperms 755 "${dir}"/jre64/bin/{java,jjs,keytool,orbd,pack200,policytool,rmid,rmiregistry,servertool,tnameserv,unpack200}
```
Comment (juippis) [inline dev-util/clion/clion-2018.2.6.ebuild]: (Opinion, but this line looks very long)

File: dev-util/clion/clion-2018.2.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Comment (juippis) [inline dev-util/clion/clion-2018.2.6.ebuild]: Also, isn't this line outdated?

File: dev-util/clion/clion-2018.2.6.ebuild (hunk 2 above)
Comment (strahlc) [reply dev-util/clion/clion-2018.2.6.ebuild]: the minus in front of the use flag disables it per default

File: dev-util/clion/clion-2018.2.6.ebuild (hunk 3 above)
Comment (strahlc) [reply dev-util/clion/clion-2018.2.6.ebuild]: In principle I agree with you, in this case the operation would become more confusing by splitting it into several lines (my opinion)

File: dev-util/clion/clion-2018.2.6.ebuild (hunk 2 above)
Comment (juippis) [reply dev-util/clion/clion-2018.2.6.ebuild]: But isn't it disabled by default? Then you can enable it with `+`. I don't think I've ever seen `-` being used there before :)

`[ebuild  N     ] dev-util/clion-2018.2.6::localhost  USE="-custom-jdk" 0 KiB`
with neutral IUSE. But once again I'm just wondering.

Comment (mgorny) [reply dev-util/clion/clion-2018.2.6.ebuild]: That `-flag` thing is an obscure historical thingy that's supposed to workaround something. You really oughtn't use it in new ebuilds.

**Comment by gyakovlev** [comment]
Sorry for delay, long holidays.

looks like it wants <=dev-util/cmake-3.13

image


I have
```sh
dev-util/cmake-3.13.0::gentoo was built with the following:
USE="ncurses qt5 system-jsoncpp -doc -emacs -test"
```
I think it should be fine, and I guess it's just a outdated upstream version constraint.
LMK if it's fine from your point of view and I'll merge, just need to clarify this small detail.

**Comment by strahlc** [comment]
@gyakovlev I've updated the PR to the latest released version of clion. But because even the new version of clion is complaining about cmake 3.13 I've also adapted the ebuild to match the upstream constraints.

**Comment by gyakovlev** [comment]
thanks! pushed.
Please don't forget to check cmake requirement next time you bump it, it'd be nice not to have it in the ebuild. hopefully upstream will test and make it work with 3.13
I also closed bug
https://bugs.gentoo.org/672100

**Pull request closed without merging**